### PR TITLE
Fix two long running wallet tests

### PR DESCRIPTION
**Pull request by philipr-za**
## Description
Changes in the transaction service protocols made these tests run longer unnecessarily. This PR fixes these tests to run in minimal necessary time.

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/wallet/tests/transaction_service/service.rs [hunk 1]
```diff
@@ -2841,10 +2837,9 @@ fn test_direct_vs_saf_send_of_tx_reply_and_finalize() {
 
     let _ = alice_outbound_service.wait_call_count(1, Duration::from_secs(60));
     let _ = alice_outbound_service.pop_call().unwrap();
-    for _ in 0..3 {
-        assert_eq!(alice_outbound_service.call_count(), 0, "Should be no more calls");
-        runtime.block_on(async { delay_for(Duration::from_secs(5)).await });
-    }
+
+    runtime.block_on(async { delay_for(Duration::from_secs(5)).await });
```
Comment (sdbondi) [inline base_layer/wallet/tests/transaction_service/service.rs:2841]: Might want to use `async_assert_eventually!` i.e poll `alice_outbound_service.call_count()` every interval until it equals the expected value or times out 

```rust
runtime.block_on(async move {
async_asssert_eventually!(
   alice_outbound_service.call_count(),
   expect = 0,
   max_attempts = 20,
   interval = Duration::from_millis(250),
);
});

**Review by sdbondi — COMMENTED** [review]
I think we need some better tooling around asynchronous expectations (probably involving https://docs.rs/tokio/0.2.21/tokio/sync/index.html#watch-channel) but `assert_async_eventually!` is the current answer to that.

File: base_layer/wallet/tests/transaction_service/service.rs (hunk 1 above)
Comment (philipr-za) [reply base_layer/wallet/tests/transaction_service/service.rs:2841]: In this case doesn't this amount to exactly the same thing? The difference to the normal usage is that we are waiting for the negative case so this will wait 5 seconds just like the original code.

Comment (sdbondi) [reply base_layer/wallet/tests/transaction_service/service.rs:2841]: Oh I see - I'd say there's probably no need to test a negative or some other way that this should be tested. But ok, in that case `async_assert_eventually!` is not correct.

**Review by sdbondi — APPROVED** [review]
LGTM
